fix(swaps): hide incompatible selection banner on quote fetch

bakoushin opened the pull request:
### Description

Observed behavior: incompatible selection banner disappears only after the new quote is fetched
Expected behavior: incompatible selection banner disappears right after another token is selected

Context: https://valora-app.slack.com/archives/C04B61SJ6DS/p1723220851805809?thread_ts=1723219373.676809&cid=C04B61SJ6DS

The same applies for other quote-dependent banners as well:
* cross-chain fee warning
* decrease spend amount for gas
* non enough balance for gas
* show missing price impact

| Before | After |
|--------|--------|
| ![before](https://github.com/user-attachments/assets/0853ff8a-01f4-4006-b6b0-dfaf2d923466) | ![after](https://github.com/user-attachments/assets/45d41d14-626e-4e08-8f4c-814a8650e41c) | 

### Test plan

Visual assessment

### Related issues

NA

### Backwards compatibility

Y

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [x] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

kathaypacific reviewed (approved):
ah thank you!! i was wondering about this the other day, should we hide all warnings and checks when there is a quote pending? it seems like all checks (apart from the first one) are on the quote - even the first one seems okay to be hidden during the quote fetch...

bakoushin commented:
@kathaypacific yes, this makes sense.

added the same condition for other quote-dependent banners:
* cross-chain fee warning
* decrease the from amount for gas
* non enough balance for gas
* show missing price impact

these 3 will keep their presence while fetching quote because they aren't dependent on quote:
* switched network (depends on token selection, potentially redundant with the advent of cross-chain swaps)
* insufficient balance (depends on balance/from amount)
* swap max amount warning (depends on balance/from amount)
